web_app_drawer/static/src/js/web_app_drawer.js

```diff
@@ -0,0 +1,287 @@
+        handleNavKeys: function(e) {
+            if (!this.isOpen){
+                return;
+            }
+            var directionCode = $.hotkeys.specialKeys[e.keyCode.toString()];
+            if (Object.keys(this.directionCodes).indexOf(directionCode) !== -1) {
+                var $link = this.findAdjacentAppLink(
+                    this.$el.find('a:first, a:focus').last(),
+                    this.directionCodes[directionCode]
+                );
+                this.selectAppLink($link);
+            } else if ($.hotkeys.specialKeys[e.keyCode.toString()] == 'esc') {
+                this.handleClickZones();
+            } else {
+                var buffer = this.handleKeyBuffer(e.keyCode);
+                this.selectAppLink(this.searchAppLinks(buffer));
+            }
+        },
+        
+        /* It adds to keybuffer, sets expire timer, and returns buffer
+         * @returns str of current buffer
+         */
+        handleKeyBuffer: function(keyCode) {
+            this.keyBuffer += String.fromCharCode(keyCode);
+            if (this.keyBufferTimeoutEvent) {
+                clearTimeout(this.keyBufferTimeoutEvent);
+            }
+            this.keyBufferTimeoutEvent = setTimeout(
+                $.proxy(this.clearKeyBuffer, this),
+                this.keyBufferTime
+            );
+            return this.keyBuffer;
+        },
+        
+        clearKeyBuffer: function() {
+            this.keyBuffer = '';
+        },
+        
+        /* It performs close actions
+         * @fires ``drawer.closed`` to the ``core.bus``
+         * @listens ``drawer.opened`` and sends to onDrawerOpen
+         */
+        onDrawerClose: function() {
+            core.bus.trigger('drawer.closed');
+            this.$el.one('drawer.opened', $.proxy(this.onDrawerOpen, this));
+            this.isOpen = false;
+        },
+        
+        /* It finds app links and register event handlers
+         * @fires ``drawer.opened`` to the ``core.bus``
+         * @listens ``drawer.closed`` and sends to :meth:``onDrawerClose``
+         */
+        onDrawerOpen: function() {
+            this.$appLinks = $('.app-drawer-icon-app').parent();
+            this.selectAppLink($(this.$appLinks[0]));
+            this.$el.one('drawer.closed', $.proxy(this.onDrawerClose, this));
+            this.setIScrollProbes();
+            core.bus.trigger('drawer.opened');
+            this.isOpen = true;
+        },
+        
+        // It sets the iScroll event handlers and probe type
+        setIScrollProbes: function() {
+            this.$el.iScroll.options.probeType = 2;
+            this.$el.iScroll.on('scroll', $.proxy(this.onIScroll, this));
+            this.$el.iScroll.on('scrollEnd', $.proxy(this.onIScroll, this));
+        },
+        
+        // It handles the iScroll Scroll events
+        onIScroll: function() {
+            var transform = (this.$el.iScroll.y) ? this.$el.iScroll.y * -1 : 0;
```

Comment:
Note to self: do something similar for the secondary nav - it should also stick to top.